### ISSUE TITLE
Plugin refinement

### ISSUE THREAD

**Issue by magurotuna** [denoland/deno_lint#908]
We already have a prototype of plugin, but it needs to be refined further to make it ready for use. This issue gives a TODO list of necessary refinements.

- making plugins writable in TypeScript, maybe powered by the internal mechanism of Deno CLI
- allowing to fetch remote modules, which also may be powered by Deno CLI
- switching from [visitor.js](https://github.com/denoland/deno_lint/blob/f7cab1e2ed19d08be367da29a565170d4dd834e4/examples/dlint/visitor.js) to [the Deno version of ast-view](https://github.com/dprint/dprint-swc-ecma-ast-view/tree/c749a035647f8943b76243e7e17f7e3a786323bc/deno)
  - can we make [this function signature](https://github.com/dprint/dprint-swc-ecma-ast-view/blob/c749a035647f8943b76243e7e17f7e3a786323bc/deno/setup.generated.ts#L5) more type-safe? `module: any` cannot be changed?
- considering how to enable plugins via config file
  - it would be good to have some mechanism to "group" additional rules, like ESLint does
  - users would write like:
```ts
import NoConsole from "./no_console.ts";
import Eqeqeq from "./eqeqeq.ts";

DenoLint.register({
  name: "mySuperPlugin",
  rules: [
    NoConsole,
    Eqeqeq,
  ] 
});
```
- writing documentation on how to implement plugins
  - at least an actual implementation example of a plugin would be fine
  - complicated examples are not really necessary at the initial stage

**Comment by bartlomieju** [comment]
This is outdated, closing in favor of https://github.com/denoland/deno/pull/27203
